src/submissions.ts

```diff
@@ -101,45 +101,48 @@ export async function createSubmission(submissionDataPayload: unknown): Promise<
     sLangProg: submissionData.answer.language,
   });
 
-  await Db.execute("insert into tm_source_codes (ID, idUser, idPlatform, idTask, sDate, sParams, sName, sSource, bSubmission) values(:idNewSC, :idUser, :idPlatform, :idTask, NOW(), :sParams, :idSubmission, :sSource, '1');", {
-    idNewSC: idNewSourceCode,
-    idUser: params.idUser,
-    idPlatform: params.idPlatform,
-    idTask: params.idTaskLocal,
-    sParams: sourceCodeParams,
-    idSubmission,
-    sSource: submissionData.answer.sourceCode
-  });
-
-  await Db.execute('insert into tm_submissions (ID, idUser, idPlatform, idTask, sDate, idSourceCode, sMode) values(:idSubmission, :idUser, :idPlatform, :idTask, NOW(), :idSourceCode, :sMode);', {
-    idSubmission,
-    idUser: params.idUser,
-    idPlatform: params.idPlatform,
-    idTask: params.idTaskLocal,
-    idSourceCode: idNewSourceCode,
-    sMode: mode,
-  });
-
-  if ('UserTest' === mode && submissionData.userTests) {
-    for (const [index, test] of submissionData.userTests.entries()) {
-      await Db.execute("insert into tm_tasks_tests (idUser, idPlatform, idTask, sGroupType, sInput, sOutput, sName, iRank, idSubmission) values(:idUser, :idPlatform, :idTask, 'Submission', :sInput, :sOutput, :sName, :iRank, :idSubmission);", {
-        idUser: params.idUser,
-        idPlatform: params.idPlatform,
-        idTask: params.idTaskLocal,
-        sInput: test.input,
-        sOutput: test.output,
-        name: test.name,
-        iRank: index,
-        idSubmission,
-      });
+  await Db.transactional(async connection => {
+    await Db.executeInConnection(connection, "insert into tm_source_codes (ID, idUser, idPlatform, idTask, sDate, sParams, sName, sSource, bSubmission) values(:idNewSC, :idUser, :idPlatform, :idTask, NOW(), :sParams, :idSubmission, :sSource, '1');", {
+      idNewSC: idNewSourceCode,
+      idUser: params.idUser,
+      idPlatform: params.idPlatform,
+      idTask: params.idTaskLocal,
+      sParams: sourceCodeParams,
+      idSubmission,
+      sSource: submissionData.answer.sourceCode
+    });
+
+    await Db.executeInConnection(connection, 'insert into tm_submissions (ID, idUser, idPlatform, idTask, sDate, idSourceCode, sMode) values(:idSubmission, :idUser, :idPlatform, :idTask, NOW(), :idSourceCode, :sMode);', {
+      idSubmission,
+      idUser: params.idUser,
+      idPlatform: params.idPlatform,
+      idTask: params.idTaskLocal,
+      idSourceCode: idNewSourceCode,
+      sMode: mode,
+    });
+
+    if ('UserTest' === mode && submissionData.userTests && submissionData.userTests.length) {
+      const valuesToInsert = [];
+      for (const [index, test] of submissionData.userTests.entries()) {
+        valuesToInsert.push({
+          idUser: params.idUser,
+          idPlatform: params.idPlatform,
+          idTask: params.idTaskLocal,
+          sInput: test.input,
+          sOutput: test.output,
+          name: test.name,
+          iRank: index,
+          idSubmission,
+        });
+      }
```

Comment:
Should be equivalent to the more readable:
```suggestion
      const valuesToInsert = submissionData.userTests.map((test,index) => ({
          idUser: params.idUser,
          idPlatform: params.idPlatform,
          idTask: params.idTaskLocal,
          sInput: test.input,
          sOutput: test.output,
          name: test.name,
          iRank: index,
          idSubmission,
      });
```
(to be checked/tested)